### PR TITLE
AO3 4704 Extend tests for the tag_wranglings controller.

### PR DESCRIPTION
**Pull request by zz9pzza**
# Pull Request Checklist
- [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
- [x] Have you read through the [contributor guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
- [x] Have you added tests for any changed functionality?
- [x] Have you added the JIRA issue number as the _first_ thing in your pull request title (eg: `AO3-1234 Fix thing`)
- [x] Have you updated the JIRA issue with the information below?
## Issue

https://otwarchive.atlassian.net/browse/AO3-4704
## Purpose

Extend tests for the tag_wranglings controller.
## Testing

No manual testing required.

### CODE REVIEW

File: features/step_definitions/generic_steps.rb [hunk 1]
```diff
@@ -2,6 +2,10 @@
   unselect(item, :from => selector)
 end
 
+When /^(?:|I )pry in to the system$/ do
+  require 'pry'
+  binding.pry
```
Comment (zz9pzza) [inline features/step_definitions/generic_steps.rb]: Ha, really no its useful

Comment (sarken) [inline features/step_definitions/generic_steps.rb]: Just wondering -- can you explain the difference between this step and the one below it, which is:
```
Then /^debug$/ do
    binding.pry
end
```
Basically, why do we need one with `require 'pry'` and one without?

Also, I think you need a blank line between these two steps.

File: features/tags_and_wrangling/tag_wrangling_more.feature [hunk 2]
```diff
@@ -141,10 +141,83 @@ Feature: Tag wrangling: assigning wranglers, using the filters on the Wranglers
        And I check the canonical option for the tag "Faye Valentine is a sweetie"
        And I check the canonical option for the tag "Ed is a sweetie"
        And I press "Wrangle"
-     Then I should see "The following tags were successfully wrangled to Cowboy Bebop: Faye Valentine is a sweetie, Ed is a sweetie Wrangle Tags for Cowboy Bebop"
+     Then I should see "The following tags were successfully wrangled to Cowboy Bebop: Faye Valentine is a sweetie, Ed is a sweetie"
        And the "Faye Valentine is a sweetie" tag should be canonical
        And the "Ed is a sweetie" tag should be canonical
 
+  Scenario: more wrangling....Freeform
```
Comment (sarken) [inline features/tags_and_wrangling/tag_wrangling_more.feature]: Could you make this a little more descriptive? What specifically are we testing here? What I see happening is we're making sure we can mass wrangle in the fandoms bin, but I can't tell:

- Why this test needs to create freeform tags
- Why this test needs to create works
- Why the test needs to create a media category

File: features/tags_and_wrangling/tag_wrangling_more.feature [hunk 3]
```diff
@@ -141,10 +141,83 @@ Feature: Tag wrangling: assigning wranglers, using the filters on the Wranglers
...
+      And I post the work "Brain Scratch" with fandom "Cowboy Bebop" with freeform "Spike Spiegel is a sweetie"
+      And I post the work "Asteroid Blues" with fandom "Cowboy Bebop" with freeform "Faye Valentine is a sweetie"
+     When the tag wrangler "lain" with password "lainnial" is wrangler of "Cowboy Bebop"
+       And I go to the tagwranglings page
+       And I follow "Fandoms by media (2)"
+       And I check the wrangling option for "Cowboy Bebop"
+       And I press "Wrangle"
+     Then I should not see "Cowboy Bebop"
+
+  Scenario: more wrangling....Relationships
```
Comment (sarken) [inline features/tags_and_wrangling/tag_wrangling_more.feature]: Again, I think we need a better description for the scenario. What I see happening is we're making sure a relationship can't be wrangled into a fandom that isn't a canonical tag, but I'm not sure

- Why the character tags are necessary
- Why the media tag is necessary
- Why the work is necessary

File: features/tags_and_wrangling/tag_wrangling_more.feature [hunk 4]
```diff
@@ -141,10 +141,83 @@ Feature: Tag wrangling: assigning wranglers, using the filters on the Wranglers
...
+      And I post the work "The course of true love never did run smooth" with fandom "October Daye Series - Seanan McGuire" with character "Tybalt (Toby Daye)" with second character "October Daye" with relationship "Toby Daye/Tybalt"
+     When the tag wrangler "lain" with password "lainnial" is wrangler of "October Daye Series - Seanan McGuire"
+       And I go to the tagwranglings page
+       And I follow "Relationships by fandom (1)"
+       And I check the wrangling option for "Toby Daye/Tybalt"
+       And I fill in "fandom_string" with "October Daye Series - Seanan McGuire"
+       And I press "Wrangle"
+     Then I should see "The following names are not canonical fandoms: October Daye Series - Seanan McGuire."
+
+  Scenario: more wrangling....Relationships canonically
```
Comment (sarken) [inline features/tags_and_wrangling/tag_wrangling_more.feature]: Same thing --  I think we need a better description for the scenario. What I see happening is we're making sure a relationship can be wrangled into a canonical fandom from the relationships bin, but I'm not sure

- Why the character tags are necessary
- Why the media tag is necessary
- Why the work is necessary

File: features/tags_and_wrangling/tag_wrangling_more.feature [hunk 5]
```diff
@@ -141,10 +141,83 @@ Feature: Tag wrangling: assigning wranglers, using the filters on the Wranglers
...
+      And I post the work "The course of true love never did run smooth" with fandom "October Daye Series - Seanan McGuire" with character "Tybalt (Toby Daye)" with second character "October 'Toby' Daye" with relationship "Toby Daye/Tybalt"
+     When the tag wrangler "lain" with password "lainnial" is wrangler of "October Daye Series - Seanan McGuire"
+       And I go to the tagwranglings page
+       And I follow "Relationships by fandom (1)"
+       And I check the wrangling option for "Toby Daye/Tybalt"
+       And I fill in "fandom_string" with "October Daye Series - Seanan McGuire"
+       And I press "Wrangle"
+     Then I should see "The following tags were successfully wrangled to October Daye Series - Seanan McGuire: Toby Daye/Tybalt"
+
+  Scenario: more wrangling....Characters can be canonical
```
Comment (sarken) [inline features/tags_and_wrangling/tag_wrangling_more.feature]: Again, I think we need a better description for the scenario -- something like "Tag wranglers can mass wrangle characters into canonical fandoms from the Characters bin." (It occurs to me 0-use tags _might_ not show up in the wrangling bin, in which case I'd suggest shortening this by posting one work with two character tags and not creating the character tags in the `Given the following typed tags exists` step.)

File: features/tags_and_wrangling/tag_wrangling_more.feature (hunk 3 above)
Comment (zz9pzza) [inline features/tags_and_wrangling/tag_wrangling_more.feature]: The truthful answer of writing a test to check the other path and collect the last line in the routines worth of coverage doesn't count ?

**Pull request merged**